routes/users.js

```diff
@@ -3,35 +3,22 @@ var router = express.Router();
 var db = require("../db.js");
 
 router.route("/")
-  .get(function (req, res) {
-    res.statusCode = 200;
-    res.json({
-      "id": 1,
-      "display_name": "BobJoe",
-      "name": "Robert Joseph"
-    })
-  })
-  .put(function(req, res) {
-    res.statusCode = 200;
-    res.end();
-  })
   .post(function (req, res) {
-    res.statusCode = 200;
-      res.json({
-      "id": 2,
-      "display_name": "Anon421235",
-      "name": "Michael Jackson"
-    })
-  })
-  .delete(function(req, res) {
-    res.statusCode = 200;
-    res.end();
+    let user = req.body;
+    db.users.insert(user).then((result) => {
+      if (result[0].affectedRows === 0) {    // If no rows are affected, send a 404.
+        res.statusCode = 404;
+        res.end();
+        return;
+      }
+       res.statusCode = 200;
+       res.json({ "id": result.insertId })
+    });
   })
 
-router
-  .route("/:id")
+router.route("/:id")
   .get(function (req, res) {
-    db.users.get(req.params.id).then(function (user) {
+    db.users.get(req.params.id).then((user) => {
       if (!user) {
         res.statusCode = 404;
```

Comment:
This should have a `res.end()` so the browser doesn't hang. Every route needs to have a `res.end()` or `res.json()` at the end